dwm1001/src/lib.rs

```diff
@@ -904,15 +908,15 @@ impl DW_IRQ {
             // Safe, as I don't believe this can interfere with the critical
             // section we're in.
             unsafe { nrf52::NVIC::unmask(Interrupt::GPIOTE); }
-            timer.enable_interrupt(nvic);
+            timer.enable_interrupt();
 
             asm::dsb();
             asm::wfi();
 
             // If we don't do this, the (probably non-existing) interrupt
             // handler will be called as soon as we exit this closure.
             nrf52::NVIC::mask(Interrupt::GPIOTE);
-            timer.disable_interrupt(nvic);
+            timer.disable_interrupt();
```

Comment:
You may want to disable the interrupt in the nvic here as well. The HAL used to do this for you.